packages/bot-web-ui/src/pages/bot-builder/quick-strategy/form-wrappers/upgraded-qs-v2/desktop-form-wrapper.tsx

```diff
@@ -0,0 +1,211 @@
+import { useFormikContext } from 'formik';
+import React from 'react';
+import { useDBotStore } from 'Stores/useDBotStore';
+import { rudderStackSendQsEditStrategyEvent } from '../../../../../analytics/rudderstack-quick-strategy';
+import { STRATEGIES } from '../../config';
+import '../../quick-strategy.scss';
+import { TFormData, TFormValues } from '../../types';
+import StrategyTabContent from '../strategy-tab-content';
+import useQsSubmitHandler from '../useQsSubmitHandler';
+import QSStepper from './qs-stepper';
+import StrategyTemplatePicker from './strategy-template-picker';
+import { QsSteps } from './trade-constants';
+
+type TDesktopFormWrapper = {
+    children: React.ReactNode;
+    current_step: QsSteps;
+    setCurrentStep: (current_step: QsSteps) => void;
+    onClickClose: () => void;
+    selected_trade_type: string;
+    setSelectedTradeType: (selected_trade_type: string) => void;
+};
+
+const QuickSelectionPanel = ({
+    selected_trade_type,
+    selected_startegy_label,
+    children,
+}: Pick<TDesktopFormWrapper, 'selected_trade_type' | 'children'> & { selected_startegy_label: string }) => (
+    <>
+        <div className='qs__selected-options'>
+            <div className='qs__selected-options__item'>
+                <Text size='xs'>{localize('Trade type')}</Text>
+                <Text size='xs' weight='bold'>
+                    {selected_trade_type}
+                </Text>
+            </div>
+            <div className='qs__selected-options__item'>
+                <Text size='xs'>{localize('Strategy')}</Text>
+                <Text size='xs' weight='bold'>
+                    {selected_startegy_label}
+                </Text>
+            </div>
+        </div>
+        <StrategyTabContent formfields={children} active_tab={'TRADE_PARAMETERS'} />
+    </>
+);
+
+const FormWrapper = observer(
+    ({
+        children,
+        current_step,
+        setCurrentStep,
+        onClickClose,
+        selected_trade_type,
+        setSelectedTradeType,
+    }: TDesktopFormWrapper) => {
+        const scroll_ref = React.useRef<HTMLDivElement & SVGSVGElement>(null);
+        const { submitForm, isValid, setFieldValue, validateForm, values } = useFormikContext<TFormValues>();
+        const { quick_strategy } = useDBotStore();
+        const { selected_strategy, onSubmit, is_stop_bot_dialog_open } = quick_strategy;
+        const { handleSubmit } = useQsSubmitHandler();
+
+        const selected_startegy_label = STRATEGIES[selected_strategy as keyof typeof STRATEGIES].label;
+        const is_selected_strategy_step = current_step === QsSteps.StrategySelect;
+
+        React.useEffect(() => {
+            if (isValid && current_step === QsSteps.StrategyVerified) {
+                setCurrentStep(QsSteps.StrategyCompleted);
+            }
+            if (!isValid && current_step === QsSteps.StrategyCompleted) {
+                setCurrentStep(QsSteps.StrategyVerified);
+            }
+        }, [isValid, current_step]);
```

Comment:
@maryia-matskevich-deriv do we need to add `current_step` dependency here? Can't we update the `setCurrentStep` based on `isValid`? I think the initial step is out of the question on this component.

Reply:
current_step is necessary to avoid situation when isValid doesn't update